pkg/router/template/router.go

```diff
@@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	config := ServiceAliasConfig{
 		Host: host,
 		Path: route.Spec.Path,
+		Hash: backendKey,
+	}
+
+	// Possibly hash the name for later use in obsuring cookie ID
+	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
```

Comment:
On 04/22/2016 06:21 PM, Clayton Coleman wrote:
>
> In pkg/router/template/router.go 
> <https://github.com/openshift/origin/pull/8615#discussion_r60810832>:
>
> > @@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
> >  	config := ServiceAliasConfig{
> >  		Host: host,
> >  		Path: route.Spec.Path,
> > +		Hash: backendKey,
> > +	}
> > +
> > +	// Possibly hash the name for later use in obsuring cookie ID
> > +	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
>
> Code used from inside the router needs to be parameterized from the 
> router cmd and passed as members. Only values that are solely used 
> within the template should be environment variables (basically, no 
> os.Getenv except in templates or in top level CLI).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8615/files/efbfa149db72f9378c1378793e9db0263996d6f0#r60810832>
>
Thanks for taking a look. Is it OK to pass the environment variables 
through the dc?I will make the changes.